NEWS.md

```diff
@@ -1,133 +1,203 @@
-   ([#3794](https://github.com/syslog-ng/syslog-ng/pull/3794))
- * `bison`: support build with bison 3.8
-   ([#3784](https://github.com/syslog-ng/syslog-ng/pull/3784))
+ * `filter`: fixed the `not` operator in `filter` expressions (regression in v3.35.1)
+
+   Reusing a filter that contains the `not` operator more than once, or
+   referencing a complex expression containing `not` might have caused invalid results
+   in the previous syslog-ng version (v3.35.1).  This has been fixed.
+   ([#3863](https://github.com/syslog-ng/syslog-ng/pull/3863))
+ * `throttle()` filter: support negation
+   ([#3863](https://github.com/syslog-ng/syslog-ng/pull/3863))
+ * `disk-buffer()`: fixed a crash which could happen in very rare cases, while a corrupted `disk-buffer` was getting replaced
+   ([#3845](https://github.com/syslog-ng/syslog-ng/pull/3845))
+ * `disk-buffer()`: fixed a memory leak issue and inconsistent buffer handling in rare cases
+   ([#3887](https://github.com/syslog-ng/syslog-ng/pull/3887))
+ * `disk-buffer()`: fixed underflowing `queued` stats counter
+   ([#3887](https://github.com/syslog-ng/syslog-ng/pull/3887))
+ * `disk-buffer()`: fixed `queued` stats were not adjusted when a disk-buffer became corrupt
+   ([#3851](https://github.com/syslog-ng/syslog-ng/pull/3851))
+ * `disk-buffer()`: fixed a disk-buffer corruption issue
+ * `amqp()`: fixed a minor error reporting problem.
+   ([#3869](https://github.com/syslog-ng/syslog-ng/pull/3869))
+ * `amqp()`: syslog-ng now drops messages that are too large to send
+   ([#3869](https://github.com/syslog-ng/syslog-ng/pull/3869))
+ * `amqp()`: fixed a crash, which happened with `librabbitmq` v0.9.0 or v0.10.0, while using the `tls()` block.
+   ([#3929](https://github.com/syslog-ng/syslog-ng/pull/3929))
+ * `file()` source: fixed invalid buffer handling when `encoding()` is used
+
+   A bug has been fixed that - under rare circumstances - could cause message
+   duplication or partial message loss when non-fixed length or less known
+   fixed-length encodings are used.
+   ([#3892](https://github.com/syslog-ng/syslog-ng/pull/3892))
+ * `syslog-ng`: fixed a SIGSEGV triggered by an incorrectly formatted "CONFIG"
+   command, received on the syslog-ng control socket.  The only known
+   implementation of the control protocol is syslog-ng-ctl itself, which always
+   sends a correct command, but anyone with access to the UNIX domain socket
+   `syslog-ng.ctl` (root only by default) can trigger a crash.
+   ([#3900](https://github.com/syslog-ng/syslog-ng/pull/3900))
+ * `credit-card-mask()`: fixed visa, mastercard and jcb card regex pattern
+   ([#3853](https://github.com/syslog-ng/syslog-ng/pull/3853))
+ * `cisco-parser()`: allow a leading dot in the timestamp (not synced clocks)
+   ([#3843](https://github.com/syslog-ng/syslog-ng/pull/3843))
+
+   A completely filled and then emptied disk-buffer may have been recognised as corrupt.
+   ([#3874](https://github.com/syslog-ng/syslog-ng/pull/3874))
```

Comment:
This belongs to "`disk-buffer()`: fixed a disk-buffer corruption issue".
Is this a bug in the news generator script?

Reply:
No, I messed it up by hand, sorry. Fixed.

Reply:
Nice catch btw!